src/components/TabBar/TabBar.jsx

```diff
@@ -44,6 +45,8 @@ const TabBar = ({ tabs, match }) => {
     setTabValue(index);
   }, [history]);
 
+  const visibleTabs = useMemo(() => tabs.filter(tab => (!tab.hidden || (tab.hidden && !tab.hidden(match)))), [tabs]);
```

Comment:
isn't the only change here moving the logic? I'm not sure why this would change the behavior.

Reply:
Possibly an issue with the previous commit https://github.com/odota/web/commit/fc10a7cfc401a3f6727271a5d17b041fea331271#diff-571eda7df4f281922d74a5ed5b6a60c7?

Reply:
Let me check again. For me it was working fine.

>isn't the only change here moving the logic? I'm not sure why this would change the behavior.

No it also changes the index while iterating through the list of tabs. That was what caused the issue in the first place since I used the "setActive" index of the unfiltered list, but the filtered list was rendered.

So the active tab was for example `3` while the actual tab index was `2` because one tab was not visible.

Moving it in front of the rendering map allows us to get the correct index. Let me check why it is not working

Reply:
Ah so the problem is fixed but the issue with initially loading the correct tab is broken.
![fldCVHrxPe](https://user-images.githubusercontent.com/6538827/67610123-ffc59180-f790-11e9-9724-c493656de566.gif)

As you can see the tab is right when you actually click around.

But on initial load the tab is wrong:
![YCgskM1WGL](https://user-images.githubusercontent.com/6538827/67610141-1e2b8d00-f791-11e9-8df0-2944209cf9d8.gif)

I'll fix the problem!

Reply:
Found the problem. A few lines above in `TabBar.jsx` I used `useEffect` to set the initial tab index without using the filtered tabs list.

Changing it to this:

```jsx
useEffect(() => {
  visibleTabs.forEach((tab, i) => {
    if (location.pathname === tab.route) setTabValue(i);
  });
}, [visibleTabs]);
```

worked fine:
![ARJom5Ey3k](https://user-images.githubusercontent.com/6538827/67610212-811d2400-f791-11e9-9b60-1a9bc6720bdb.gif)